src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterApplication.java

```diff
@@ -57,7 +72,14 @@ public final void initialize(Bootstrap<MatchingServiceAdapterConfiguration> boot
         //
         // https://github.com/google/guice/issues/357
         GuiceBundle<MatchingServiceAdapterConfiguration> guiceBundle = defaultBuilder(MatchingServiceAdapterConfiguration.class)
-                .modules(new MatchingServiceAdapterModule())
+                .modules(new MatchingServiceAdapterModule(), new AbstractModule() {
```

Comment:
Is it the intention that we use this bundle in the hub too? In which case, could we define this _module_ in the library and then just pass the `MetadataResolverBundle` instance to it, meaning we can re-use this module code too?

Reply:
Saml-engine, and soon the rest of the Hub, are now using this. Different services need to use the objects in different ways so there isn't an easy generic solution for that right now. Some services require named dependencies while others don't. Other services will only need to bind a few things. Introducing a generic module for this is hard. Especially, as this bundle is not just used for reading the Federation metadata (e.g. VSP).

Additionally, the intent was to remove Guice as a dependency on the saml-metadata-bindings code entirely. That was complicated by the new eIDAS code, but that is still the intent.

Finally, saml-metadata-bindings is currently providing a Module getter for just the metadata resolver. However, this doesn't seem to work with our applications. I think perhaps because of differences between the guice dependencies used by the libraries and different services.